Looms.PoS/Looms.PoS.Application/Features/Reservation/Commands/CreateReservation/CreateReservationRequestValidator.cs

```diff
@@ -0,0 +1,74 @@
+using FluentValidation;
+using Looms.PoS.Application.Models.Requests.Reservation;
+using Looms.PoS.Application.Utilities.Validators;
+using Looms.PoS.Application.Utilities;
+using Looms.PoS.Domain.Interfaces;
+
+namespace Looms.PoS.Application.Features.Reservation.Commands.CreateReservation;
+
+public class CreateReservationRequestValidator : AbstractValidator<CreateReservationRequest>
+{
+    public CreateReservationRequestValidator(IServicesRepository servicesRepository, IReservationsRepository reservationsRepository)
+    {
+        RuleFor(x => x.CustomerName)
+            .Cascade(CascadeMode.Stop)
+            .MustAsync(async (request, customerName, cancellation) =>
+            {
+                var appointmentTime = DateTimeHelper.ConvertToUtc(request.AppointmentTime);
+                var existingReservations = await reservationsRepository.GetReservationsByCustomerAndTimeAsync(customerName, appointmentTime);
+                var existingReservation = existingReservations.FirstOrDefault();
+                return existingReservation is null;
+            })
+            .WithMessage("An appointment for the same customer at the same time already exists.");
+
+        RuleFor(x => x.AppointmentTime)
+            .Cascade(CascadeMode.Stop)
+            .MustBeValidDateTime()
+            .MustAsync(async (request, dateString, cancellation) =>
+            {
+                var service = await servicesRepository.GetAsync(Guid.Parse(request.ServiceId));
+                var business = service.Business;
+                var appointmentTime = DateTimeHelper.ConvertToUtc(dateString);
+                return appointmentTime.Hour >= business.StartHour && appointmentTime.Hour < business.EndHour;
```

Comment:
End hour check should include service duration

Reply:
changed it